uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -2408,6 +2408,11 @@ public function testRecalculateExpiredBalanceChangesWhenAbsenceTypeOfLeaveReques
   public function testCalculateAmountForDateForAContactWithWorkPatternHavingMultipleWeeks() {
     $periodStartDate = new DateTime('2016-01-01');
 
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2016-12-31')
+    ]);
```

Comment:
Yes absence period needs to be present when creating public holiday with past dates. See here [Here](https://github.com/civicrm/civihr/blob/c3fb4d1dca6a337b6f54d2337299792e6164c544/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php#L877-L877). There did not to be much performance difference as I ran tests both on my branch and also on the la-milestone-3 branch.